### PR TITLE
[PERFORMANCE] Various small enhancements detected thanks to flame graphs

### PR DESCRIPTION
**Pull request by chibenwa**
Screenshot from 2021-05-16 17-45-01

Here we can see a call to `MessageResultImpl::getMessageId` responsible of `0.78%` of the total allocated memory... Rationals are that and intermediate (uneeded) on the fly messageMetadata is used, which includes an (innefficient) flag copy.

Screenshot from 2021-05-16 17-51-05

Here is a flame graph capture showing the costs of initializing JMAP routes for each requests.

### CODE REVIEW

**Comment by chibenwa** [comment]
Screenshot from 2021-05-16 22-41-50

0.26% CPU resources on formatting a validation message...

**Comment by chibenwa** [comment]
Screenshot from 2021-05-17 07-46-15

0.56% running an inefficient algorithm to state which mailboxes have kids, in O(n2). While the percentage is low acting on it with likely improve p99 for `Mailbox/get` with many mailboxes and is thus worth writing.

**Review by Arsnael — APPROVED** [review]

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/FlagsFactory.java [hunk 1]
```diff
@@ -60,10 +60,7 @@ private static Flags asFlags(MailboxMessage mailboxMessage, String[] userFlags)
     }
 
     public static Flags createFlags(MailboxMessage mailboxMessage, String[] userFlags) {
-        return builder()
-            .flags(asFlags(mailboxMessage, userFlags))
-            .addUserFlags(userFlags)
-            .build();
+        return asFlags(mailboxMessage, userFlags);
```
Comment (rouazana) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/FlagsFactory.java:63]: why not removing createFlags and replace it by asFlags then?

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
I confirm the last fixups regarding "computing metadata" do not cost performance.

**Review by Arsnael — APPROVED** [review]

**Review by jeantil — APPROVED** [review]

**Pull request merged**